resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/SparkReadinessWatcher.scala

```diff
@@ -30,12 +30,12 @@ private[spark] class SparkReadinessWatcher[T <: HasMetadata] extends Watcher[T]
 
   override def eventReceived(action: Action, resource: T): Unit = {
     if ((action == Action.MODIFIED || action == Action.ADDED) &&
-        Readiness.isReady(resource)) {
+        Readiness.getInstance().isReady(resource)) {
```

Comment:
do we need to pull `Readiness.getInstance()` as private variable? Is it cheap to call `Readiness.getInstance()` every call of `eventReceived`?